bridge-token-factory/src/fee.rs

```diff
@@ -0,0 +1,661 @@
+    if let Some(lower_bound) = fee_bounds.lower_bound {
+        if fee_amount < lower_bound.0 {
+            return lower_bound.0;
+        }
+    }
+
+    if let Some(upper_bound) = fee_bounds.upper_bound {
+        if fee_amount > upper_bound.0 {
+            return upper_bound.0;
+        }
+    }
+
+    fee_amount
+}
+
+#[near_bindgen]
+impl BridgeTokenFactory {
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    #[payable]
+    pub fn set_deposit_fee(
+        &mut self,
+        token: EthAddressHex,
+        fee_percentage: U128,
+        lower_bound: Option<U128>,
+        upper_bound: Option<U128>,
+    ) -> Promise {
+        self.deposit_fee.insert(
+            &token.0,
+            &Fee {
+                fee_percentage,
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+
+        require!(env::attached_deposit() >= self.bridge_token_storage_deposit_required);
+        ext_bridge_token::ext(self.get_bridge_token_account_id(&token))
+            .with_static_gas(TOKEN_STORAGE_DEPOSIT_GAS)
+            .with_attached_deposit(env::attached_deposit())
+            .storage_deposit(None, None)
+    }
+
+    // Fee should be added as per: 10% -> 0.1 = 0.1*10^6 with proper fee amount bounds
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_per_silo(
+        &mut self,
+        silo_account_id: AccountId,
+        token: Option<EthAddressHex>,
+        fee_percentage: U128,
+        lower_bound: Option<U128>,
+        upper_bound: Option<U128>,
+    ) {
+        self.deposit_fee_per_silo.insert(
+            &get_silo_fee_map_key(&silo_account_id, token.as_ref()),
+            &Fee {
+                fee_percentage,
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+    }
+
+    // Fee should be added as per: 10% -> 0.1 = 0.1*10^6 with proper fee amount bounds
+    #[access_control_any(roles(Role::FeeSetter))]
+    #[payable]
+    pub fn set_withdraw_fee(
+        &mut self,
+        token: EthAddressHex,
+        fee_percentage: U128,
+        lower_bound: Option<U128>,
+        upper_bound: Option<U128>,
+    ) -> Promise {
+        self.withdraw_fee.insert(
+            &token.0,
+            &Fee {
+                fee_percentage,
+                bounds: FeeBounds {
+                    lower_bound,
+                    upper_bound,
+                },
+            },
+        );
+
+        require!(env::attached_deposit() >= self.bridge_token_storage_deposit_required);
+        ext_bridge_token::ext(self.get_bridge_token_account_id(&token))
+            .with_static_gas(TOKEN_STORAGE_DEPOSIT_GAS)
+            .with_attached_deposit(env::attached_deposit())
+            .storage_deposit(None, None)
```

Comment:
Same comment as for `set_deposit_fee` moreover, probable we already called `storage_deposit` for this token from `set_deposit_fee` function.